Index: neovim-0.10.0/src/nvim/ex_cmds.lua
===================================================================
--- neovim-0.10.0.orig/src/nvim/ex_cmds.lua	2024-05-16 17:22:58.176811996 +0200
+++ neovim-0.10.0/src/nvim/ex_cmds.lua	2024-05-16 17:23:05.255828571 +0200
@@ -1,4 +1,4 @@
-local bit = require 'bit'
+local bit = require 'bit32'
 
 local M = {}
 
Index: neovim-0.10.0/src/nvim/generators/gen_ex_cmds.lua
===================================================================
--- neovim-0.10.0.orig/src/nvim/generators/gen_ex_cmds.lua	2024-05-16 17:22:58.177811999 +0200
+++ neovim-0.10.0/src/nvim/generators/gen_ex_cmds.lua	2024-05-16 17:23:05.256828573 +0200
@@ -10,7 +10,7 @@ local defsfname = autodir .. '/ex_cmds_d
 local enumfile = io.open(enumfname, 'w')
 local defsfile = io.open(defsfname, 'w')
 
-local bit = require 'bit'
+local bit = require 'bit32'
 local ex_cmds = require('ex_cmds')
 local defs = ex_cmds.cmds
 local flags = ex_cmds.flags
